langchain/utilities/arxiv_search.py

```diff
@@ -0,0 +1,138 @@
+
+        try:
+            import arxiv
+        except ImportError:
+            raise ImportError(
+                "arxiv is not installed. " "Please install it with `pip install arxiv`"
+            )
+        return values
+
+    def run(self, query: str) -> str:
+        """Run query through ArXiv Search and parse result."""
+        papers = self.results(query)
+
+        # Representation of a list of papers in a string:
+        # """
+        # Title: a-title
+        # Authors: a, b, c
+        # Abstract: an-abstract
+        #
+        #
+        # Title: another-title...
+        # """
+        if len(papers) == 0:
+            return "No good Arxiv Search Result was found"
+
+        results = ""
+        for paper in papers:
+            result = f"Title: {paper['title'].strip()}\n"
+            result += f"Authors: {_get_authors_str(paper['authors']).strip()}\n"
+            result += f"Abstract: {paper['abstract'].strip()}\n"
+            result += "\n\n"
+            results += result.strip()
+
+        return results
+
+    def results(self, query: str) -> List[dict]:
+        """Return results from ArXiv."""
+        return self._parse_results(self._arxiv_search_results(query))
+
+    def _parse_results(self, results: Iterator) -> List[dict]:
+        papers = []
+
+        try:
+            result = next(results)
+
+            paper_id = result.get_short_id()
+            paper_title = result.title
+            paper_url = result.entry_id
+            paper_abstract = result.summary.replace("\n", " ")
+            paper_authors = [str(author) for author in result.authors]
+            paper_first_author = _get_authors_str(result.authors, first_author=True)
+            primary_category = result.primary_category
+            publish_time = result.published.date()
+            update_time = result.updated.date()
+            comments = result.comment
+
+            paper = {
+                "id": paper_id,
+                "title": paper_title,
+                "url": paper_url,
+                "abstract": paper_abstract,
+                "authors": paper_authors,
+                "first_author": paper_first_author,
+                "primary_category": primary_category,
+                "publish_time": publish_time,
+                "update_time": update_time,
+                "comments": comments,
+            }
+            papers.append(paper)
+        except (
+            StopIteration
+        ):  # See: https://docs.python.org/3/library/stdtypes.html#iterator.__next__
+            pass  # No results found
+
+        return papers
+
+    def _arxiv_search_results(self, query: str) -> Iterator:
+        """Search arXiv for query and return results."""
+        from arxiv import Search
+
+        search_results = Search(
+            query=query,
+            max_results=self.max_results,
+            sort_by=self.sort_by,
+            sort_order=self.sort_order,
+        )
+
+        return search_results.results()
```

Comment:
It's nice to have a helper folding the `self` parameters into a search, but I'd leave it up to the caller when/how to resolve the generator.

```suggestion
    def _search(self, query: str) -> Iterator:
        """Construct an arXiv search for query with stored parameters."""
        from arxiv import Search
        return Search(
            query=query,
            max_results=self.max_results,
            sort_by=self.sort_by,
            sort_order=self.sort_order
        )
```